tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
```

Comment:
> I _definitely_ do not like that `scrollIntoView` is written the same as the proposed standard but does not take the same arguments.

That's a comment on the actual implementation, right? Because all the current PR does is adding some tests. The actual code was added in PR https://github.com/hypothesis/h/pull/1908.

Anyway, I am happy to refactor this in another PR. So basically, are you proposing to implement [this API](https://developer.mozilla.org/en-US/docs/Web/API/Element.scrollIntoView) (or something very close to it), on `Highlight`'s and `Anchor`'s `scrollIntoView()` method - right?